### PR TITLE
Updated lighting, smart plugs/outlets and switches and controls device types xml according to specs

### PR DESCRIPTION
**Pull request by jadhavrohit924**
#### Problem
What is being fixed?  Examples:
* Lighting , smart plugs/outlets and switches-and-controls device types xml was not matching with specs

#### Change overview
Updated lighting , smart plugs/outlets and switches-and-controls device types xml according to specs

#### Testing
How was this tested? (at least one bullet point required)
* Verified that the xml is in sync with the specification.

### CODE REVIEW

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 1]
```diff
@@ -323,7 +323,7 @@ limitations under the License.
                 <requireAttribute>CLIENT_LIST</requireAttribute>
                 <requireAttribute>PARTS_LIST</requireAttribute>
             </include>
-            <include cluster="Binding" client="true" server="false" clientLocked="false" serverLocked="true">
+            <include cluster="Binding" client="false" server="false" clientLocked="true" serverLocked="false">
```
Comment (PSONALl) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:326]: Binding client should be completely removed

Comment (PSONALl) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:326]: As all lighting devices come under Simple device types and they are not Client device types .. we should have a binding server as a mandatory cluster and the Binding client shouldn't be there,
So here server should be set to true

Comment (jadhavrohit924) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:326]: Binding server is mandatory when device types is Simple and Client. Here Occupancy Sensing client cluster is optional. So I set Binding client turned off and user CANNOT turn it on, while Binding server turned off but user CAN turn it on.

**Review by dhrishi — APPROVED** [review]

**Review by PSONALl — APPROVED** [review]

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 2]
```diff
@@ -1133,7 +1133,7 @@ limitations under the License.
                 <requireCommand>RemoveAllGroups</requireCommand>
                 <requireCommand>AddGroupIfIdentifying</requireCommand>
             </include>
-            <include cluster="Scenes" client="true" server="true" clientLocked="true" serverLocked="true">
+            <include cluster="Scenes" client="true" server="false" clientLocked="true" serverLocked="true">
                 <requireAttribute>SCENE_COUNT</requireAttribute>
```
Comment (jadhavrohit924) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:1137]: Should we have attributes in client clusters ? @dhrishi

Comment (dhrishi) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:1137]: @tecimovic @bzbarsky-apple Do you know why attributes are present under client clusters in the XML? Should they be removed?

**Review by woody-apple — APPROVED** [review]

**Comment by dhrishi** [comment]
@woody-apple @bzbarsky-apple Can we merge this?

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml (hunk 2 above)
Comment (bzbarsky-apple) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:1137]: Attributes in client clusters don't really make sense.  Please do a followup to remove that @jadhavrohit924 ?

**Comment by bzbarsky-apple** [comment]
@dhrishi there was an unresolved conversation that needed to be resolved...

**Comment by dhrishi** [comment]
>@dhrishi there was an unresolved conversation that needed to be resolved...

Hi @bzbarsky-apple We will still be working on other devices, as we had discussed previously. @jadhavrohit924 can add the changes in follow-up PR. Thanks!

**Comment by bzbarsky-apple** [comment]
Sure; I'm just pointing out why the PR had not merged.  When I resolved the conversation, it auto-merged.

**Comment by dhrishi** [comment]
> Sure; I'm just pointing out why the PR had not merged. When I resolved the conversation, it auto-merged.

Ohh okay.

**Comment by dhrishi** [comment]
@bzbarsky-apple Filed an issue for this: https://github.com/project-chip/connectedhomeip/issues/21540. @jadhavrohit924 will put up a PR soon

**Pull request merged**